src/client/auth.js

```diff
@@ -1,16 +1,67 @@
 const UUID = require('uuid-1345')
 const yggdrasil = require('yggdrasil')
+const fs = require('mz/fs')
+const promisify = require('es6-promisify')
+const mkdirp = promisify(require('mkdirp'))
 
-module.exports = function (client, options) {
+module.exports = async function (client, options) {
   const yggdrasilClient = yggdrasil({ agent: options.agent, host: options.authServer || 'https://authserver.mojang.com' })
-  const clientToken = options.clientToken || (options.session && options.session.clientToken) || UUID.v4().toString()
+  const clientToken = (options.profilesFolder && (await getLP()).clientToken) || options.clientToken || (options.session && options.session.clientToken) || UUID.v4().toString().replace("-", "")
   const skipValidation = false || options.skipValidation
   options.accessToken = null
-  options.haveCredentials = options.password != null || (clientToken != null && options.session != null)
+  options.haveCredentials = options.password != null || (clientToken != null && options.session != null) || options.profilesFolder != null
+
+  async function getLP () { // get launcher profile
+    let data
+    try {
+      data = JSON.parse(await fs.readFile(options.profilesFolder + '/launcher_profiles.json', 'utf8'))
+    } catch (err) {
+      await mkdirp(options.profilesFolder)
+      await fs.writeFile(options.profilesFolder + '/launcher_profiles.json', '')
+      data = { authenticationDatabase: {} }
+    }
+    return data
+  }
 
   if (options.haveCredentials) {
     // make a request to get the case-correct username before connecting.
-    const cb = function (err, session) {
+    const cb = async function (err, session) {
+      if (options.profilesFolder) {
+        let auths = await getLP()
+        let lowerUsername = options.username.toLowerCase()
+        let profile = Object.keys(auths.authenticationDatabase).find(key => 
+          auths.authenticationDatabase[key].username.toLowerCase() == lowerUsername
+          || Object.values(auths.authenticationDatabase[key].profiles)[0].displayName.toLowerCase() == lowerUsername
```

Comment:
`Object.values(auths.authenticationDatabase[key].profiles)[0]` why it's only the value of the first key here ?

Reply:
I think there's only ever 1 profile for the minecraft launcher

Reply:
Per account, that is. You can still have multiple accounts but for now those all have 1 profile.

Reply:
:+1: